packages/@aws-cdk/aws-codestarnotifications/test/notification-rule.test.ts

```diff
@@ -0,0 +1,196 @@
+import '@aws-cdk/assert-internal/jest';
+import * as cdk from '@aws-cdk/core';
+import * as notifications from '../lib';
+import {
+  FakeCodeBuild,
+  FakeCodePipeline,
+  FakeSlackTarget,
+  FakeSnsTopicTarget,
+} from './helpers';
+
+describe('Rule', () => {
+  let stack: cdk.Stack;
+
+  beforeEach(() => {
+    stack = new cdk.Stack();
+  });
+
+  test('created new notification rule with source', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: project,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+    });
+  });
+
+  test('created new notification rule with all parameters in constructor props', () => {
+    const project = new FakeCodeBuild();
+    const slack = new FakeSlackTarget();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      notificationRuleName: 'MyNotificationRule',
+      detailType: notifications.DetailType.FULL,
+      events: [
+        'codebuild-project-build-state-succeeded',
+        'codebuild-project-build-state-failed',
+      ],
+      source: project,
+      targets: [slack],
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRule',
+      DetailType: 'FULL',
+      EventTypeIds: [
+        'codebuild-project-build-state-succeeded',
+        'codebuild-project-build-state-failed',
+      ],
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Targets: [
+        {
+          TargetAddress: 'arn:aws:chatbot::1234567890:chat-configuration/slack-channel/MySlackChannel',
+          TargetType: 'AWSChatbotSlack',
+        },
+      ],
```

Comment:
Same note about the random hard-coded ARNs being present in test assertions here.

Reply:
Already adjusted.